internal/proxy/proxy.go

```diff
@@ -3,20 +3,54 @@ package proxy
 
 import (
 	"context"
+	"os"
+	"path/filepath"
 
 	log "github.com/sirupsen/logrus"
+	"github.com/ubuntu/decorate"
 )
 
 // Proxy represents a proxy manager.
-type Proxy struct{}
+type Proxy struct {
+	proxySettings []proxySetting
+	envConfigPath string
+}
 
 // DryRun is a context key to indicate that we should not apply the proxy
 // settings. It is intended to be used in tests only.
 var DryRun = struct{}{}
 
+type options struct {
+	root          string
+	envConfigPath string
+}
+type option func(*options)
+
+const confHeader = "### This file was generated by ubuntu-proxy-manager - manual changes will be overwritten"
+
 // New returns a new instance of a proxy manager.
-func New(ctx context.Context, http, https, ftp, socks, no, mode string) (*Proxy, error) {
-	return &Proxy{}, nil
+func New(ctx context.Context, http, https, ftp, socks, no, mode string, args ...option) (p *Proxy, err error) {
+	defer decorate.OnError(&err, "couldn't create proxy manager")
+
+	// Set default options
+	opts := options{
+		root:          "/",
+		envConfigPath: "etc/environment.d/99ubuntu-proxy.conf",
```

Comment:
yes, it’s just that ubuntu-proxy seems to be very specific to ubuntu, while we want to should in the filename the source of it. Maybe it could be directly ubuntu-proxy-manager.conf?